response_operations_ui/views/collection_exercise.py

```diff
@@ -190,6 +190,12 @@ def view_collection_exercise(short_name, period):
     )
 
 
+def _validate_exercise(exercise: dict, period: str, short_name: str):
```

Comment:
is this a preferable approach to doing a try catch within view_sample_ci_summary?

Reply:
I just copied an existing pattern and rationalised it. However I hadn't noticed this isn't correct and isn't working as expected, get_collection_exercises_by_survey is raising a uncaught ApiError into the view, darn

Reply:
Muddy puddle time, the code needs a refactor in this area. get_collection_exercise_by_period is the main issue, get_collection_exercises_by_survey does raise an uncaught exception, but it isn't dreadful. Its get_collection_exercise_by_period that should encapsulate the failing of a period to be found. Unfortunately it's used in 8 places, but all for the same thing, so could be rationalised. I'm going to run away and hide from this one and create a different story to correct this muddle